### PR TITLE
fix: Make sure to use current pid to verify Camel app status

### PR DESCRIPTION
**Pull request by gansheer**
## Bug

The integration tests `org.citrusframework.camel.yaml.JBangTest.shouldLoadCamelActions` and `org.citrusframework.camel.xml.JBangTest.shouldLoadCamelActions` fails on my local environement.
The Camel Jbang CamelRunIntegrationAction sets the wrong process pid from a temporary descendant process in the context. This descendant process is not the Camel Jbang run process. When the CamelVerifyIntegrationAction use the pid from the context object and keeps waiting for the integration process to be Running. 
The descendant process is not active very long.

```
Processing 19951 command: Optional[/usr/bin/bash -c jbang -Dgreeting="Hello Camel" camel@apache/camel run --name hello-yaml --verbose /home/user1/work/sources/gansheer/citrus/endpoints/citrus-camel/target/test-classes/org/citrusframework/camel/integration/route.yaml]
Processing descendant 19963 command: Optional[/usr/lib/jvm/java-21-openjdk-21.0.5.0.11-1.fc41.x86_64/bin/java -classpath /home/user1/.jbang/bin/jbang.jar dev.jbang.Main -Dgreeting=Hello Camel camel@apache/camel run --name hello-yaml --verbose /home/user1/work/sources/gansheer/citrus/endpoints/citrus-camel/target/test-classes/org/citrusframework/camel/integration/route.yaml]
2024-12-17 09:41:01.372|INFO |main|CamelRunIntegrationAction - Started Camel integration 'hello-yaml' (19963)
2024-12-17 09:41:01.382|INFO |main|CamelRunIntegrationAction - Waiting for the Camel integration 'hello-yaml' (19963) to be running ...
2024-12-17 09:41:02.344|INFO |main|CamelVerifyIntegrationAction - Camel JBang version: 4.9.0

2024-12-17 09:41:02.344|INFO |main|CamelVerifyIntegrationAction - Verify Camel integration 'hello-yaml' ...
2024-12-17 09:41:02.344|INFO |main|INTEGRATION_STATUS - Waiting for Camel integration 'hello-yaml' to be in state 'Running'
2024-12-17 09:41:04.263|INFO |main|CamelVerifyIntegrationAction - 
  PID   NAME        READY  STATUS   AGE  TOTAL  REMOTE  FAIL  INFLIGHT   
 19951  hello-yaml   1/1   Running   3s      0       0   0/0       0/0   


2024-12-17 09:41:06.560|INFO |main|CamelVerifyIntegrationAction - Waiting for Camel integration 'hello-yaml' to be in state 'Running'- retry in 2000 ms
2024-12-17 09:41:08.353|INFO |main|CamelVerifyIntegrationAction - 
  PID   NAME       READY  STATUS   AGE  TOTAL  REMOTE  FAIL  INFLIGHT   
 19951  hello-yaml   1/1   Running   7s      4       0   0/0       0/0 
```

## Fix

* Use the parent process pid to identify the running Camel JBang application
* Camel JBang pid may change on Unix OS due to descendant process being launched
* Racing conditions may lead to parent pid being stored in test context
* Make sure to also search for descendant process ids when verifying the Camel integration app status

### CODE REVIEW

File: endpoints/citrus-camel/src/main/java/org/citrusframework/camel/actions/CamelRunIntegrationAction.java [hunk 1]
```diff
@@ -128,7 +131,24 @@ public void doExecute(TestContext context) {
...
+            for (int i = 0; i < maxAttempts; i++) {
+                String log = pao.getOutput();
+                if (log.length() > 0) {
+                    break;
+                }
+
+                try {
+                    Thread.sleep(delayBetweenAttempts);
+                } catch (InterruptedException e) {
+                    logger.warn("Interrupted while waiting for Camel integration '%s' first log".formatted(name), e);
```
Comment (bbortt) [inline endpoints/citrus-camel/src/main/java/org/citrusframework/camel/actions/CamelRunIntegrationAction.java:146]:
```suggestion
                    logger.warn("Interrupted while waiting for Camel integration '{}' first log", name, e);
```

please use the `logger`-internal formatting.

**Review by bbortt — CHANGES_REQUESTED** [review]
hello and thanks for the attempted fix. there are probably differences in local execution vs pipeline. this currently doesn't run: https://github.com/citrusframework/citrus/pull/1284/checks#step:6:54177.

additionally, I've left one remark on your code.

**Comment by christophd** [comment]
I have created another branch with potential fix: https://github.com/citrusframework/citrus/tree/fix_pid_camel_jbang_run

@gansheer can you please checkout this branch and see if the fix works on your machine?

I have added some logic in the Camel JBang verification test action to also look for descendant process ids. PR is here: https://github.com/citrusframework/citrus/pull/1285

Let's see what GH actions results do report for this fix

**Comment by gansheer** [comment]
Merge in https://github.com/citrusframework/citrus/pull/1286.

**Pull request closed without merging**